### PR TITLE
Docs for OData AspNet WebApi #2191 - Passing OData query options in request body

### PR DESCRIPTION
**Pull request by gathogojr**
Adding docs for OData AspNet WebApi [#2191](https://github.com/OData/WebApi/pull/2191) - Passing OData query options in request body

### CODE REVIEW

File: Odata-docs/webapi/query-options-in-request-body.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+date:   2020-09-28 07:49:00
+
+ms.date: 9/28/2020
+---
+# Passing OData query options in request body
+**Applies To**:[!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-core-v7.5.md)][!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-v7.5.md)]
+
+The query options part of an OData URL can be quite long, potentially exceeding the maximum length that many hosting environments (including IIS) impose. This can limit the client's ability to request the exact set of data they are interested in, especially when using explict select statements.
+
+OData AspNet WebApi V7.5 introduced support for passing query options in the request body. This is achieved by prepating a POST request with the query options part of the URL in the body and sending that to an endpoint comprising of the resource path appended with `/$query` - e.g. http://ServiceRoot/Movies/$query. The `Content-Type` header should be set to `text/plain`.
```
Comment (ElizabethOkerio) [inline Odata-docs/webapi/query-options-in-request-body.md:12]: prepating? can it be a simple/common word.

**Review by ElizabethOkerio — APPROVED** [review]

File: Odata-docs/webapi/query-options-in-request-body.md (hunk 1 above)
Comment (gathogojr) [reply Odata-docs/webapi/query-options-in-request-body.md:12]: The word should be "preparing". I'll fix and push the change

File: Odata-docs/webapi/query-options-in-request-body.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+ms.date: 9/28/2020
+---
+# Passing OData query options in request body
+**Applies To**:[!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-core-v7.5.md)][!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-v7.5.md)]
+
+The query options part of an OData URL can be quite long, potentially exceeding the maximum length that many hosting environments (including IIS) impose. This can limit the client's ability to request the exact set of data they are interested in, especially when using explict select statements.
+
+OData AspNet WebApi V7.5 introduced support for passing query options in the request body. This is achieved by preparing a POST request with the query options part of the URL in the body and sending that to an endpoint comprising of the resource path appended with `/$query` - e.g. http://ServiceRoot/Movies/$query. The `Content-Type` header should be set to `text/plain`. 
+
+In the rest of this page, we demonstrate how the feature works by building a simple OData service
```
Comment (habbes) [inline Odata-docs/webapi/query-options-in-request-body.md:16]: Maybe you could show an example of request body before jumping to create a project. That way if someone is in a hurry, they can just test the request body and if it works for them, they ignore the rest of the docs.

**Comment by habbes** [comment]
@gathogojr is it worthwhile to add a section on custom parsers? Or is that beyond the scope of this article?

**Comment by gathogojr** [comment]
> @gathogojr is it worthwhile to add a section on custom parsers? Or is that beyond the scope of this article?

@habbes Took care of this by adding a page and **Web API Extensibility** https://github.com/gathogojr/OData-docs/blob/docs/2191-support-long-url/Odata-docs/webapi/custom-query-options-parser.md

**Review by habbes — APPROVED** [review]
Maybe you should mention that it's possible to customize parsers and provide a link to the relevant article.

**Comment by gathogojr** [comment]
> Maybe you should mention that it's possible to customize parsers and provide a link to the relevant article.

Added

**Pull request merged**